educational-core/src/com/jetbrains/edu/learning/codeforces/courseFormat/CodeforcesTask.kt

```diff
@@ -105,16 +107,59 @@ open class CodeforcesTask : Task() {
       return task
     }
 
+    fun codeforcesSubmitLink(task: Task): String {
+      val course = task.course as CodeforcesCourse
+      return "${course.getContestUrl()}/${CODEFORCES_SUBMIT}?locale=${course.languageCode}" +
+             (codeforcesProgramTypeId(course)?.let { "&programTypeId=$it" } ?: "") +
+             "&submittedProblemIndex=${task.name.substringBefore(".")}"
+    }
+
     fun codeforcesTaskLink(task: Task): String {
       val course = task.course as CodeforcesCourse
       return "${course.getContestUrl()}/problem/${task.name.substringBefore(".")}?locale=${course.languageCode}"
     }
 
+    private fun codeforcesProgramTypeId(course: CodeforcesCourse): Int? {
+      val languageID = course.languageID
+      val languageVersion = course.languageVersion
+      return when {
+        GO == languageID -> GO_TYPE_ID
+        JAVA == languageID && "8" == languageVersion-> JAVA_8_TYPE_ID
+        JAVA == languageID && "11" == languageVersion -> JAVA_11_TYPE_ID
+        JAVASCRIPT == languageID -> JAVASCRIPT_TYPE_ID
+        KOTLIN == languageID -> KOTLIN_TYPE_ID
+        "Node.js" == languageID -> NODEJS_TYPE_ID
```

Comment:
We don't need `Node.js` and `PyPy` at this time, so better to remove them. 
Though `ProgramTypeId` (lines 155-157) for them is worth having, let's leave them and annotate them as `@Suppress("UNUSED_VARIABLE")`?

Reply:
I think in this case just removing them will make code cleaner. It could be easily recovered if needed.